chainer/function_hooks/timer.py

```diff
@@ -61,3 +81,38 @@ def backward_postprocess(self, function, in_data, out_grad):
     def total_time(self):
         """Returns total elapsed time in seconds."""
         return sum(t for (_, t) in self.call_history)
+
+    def summary(self):
+        """A summary of time profiling in functions.
+
+        Returns:
+            A summarized dictionary whose keys are function names and
+            values are dictionaries of `elapsed_time` and `occurrrence`.
+        """
+        summary = {}
+        for _function, elapsed_time in self.call_history:
+            function_name = _function.__class__.__name__
+            if function_name not in summary:
+                summary[function_name] = {"elapsed_time": 0, "occurrence": 0}
+            record = summary[function_name]
+            record["elapsed_time"] += elapsed_time
+            record["occurrence"] += 1
+        return summary
+
+    def _humanized_time(self, second):
+        """Return a human readable time."""
+        for unit in ['sec', 'ms', 'us']:
+            if second >= 1:
+                return "%3.2f%s" % (second, unit)
+            second *= 1000.0
+        return "%.2f%s" % (second, 'ns')
+
+    def print_summary(self, sep='\t', end='\n', file=sys.stdout):
+        """Print a summary of time profiling in functions."""
+        six.print_("FunctionName", "ElapsedTime", "Occurrence",
+                   sep=sep, end=end, file=file)
```

Comment:
I think using tabs for aligninment is not a good idea, because the appearance will depend on environment.
How about using spaces by counting maximum length manually?
(You can refer to `training.extensions.print_report.PrintReport` as a similar example)